plone/app/contenttypes/browser/templates/newsitem.pt

```diff
@@ -9,10 +9,10 @@
 
 <metal:content-core fill-slot="content-core">
 <metal:content-core define-macro="content-core"
-                    tal:define="toc context/table_of_contents|nothing;">
+                    tal:define="toc python: getattr(context, 'table_of_contents', False)">
 
   <section id="section-text">
-    <div id="parent-fieldname-text" tal:condition="context/text" tal:content="structure python:context.text.output_relative_to(view.context)" tal:attributes="class python: toc and 'pat-autotoc' or ''">
+    <div id="parent-fieldname-text" tal:condition="python: getattr(context, 'text', False)" tal:content="structure python:context.text.output_relative_to(view.context)" tal:attributes="class python: toc and 'pat-autotoc' or ''">
```

Comment:
The old code was `context/text` (without) the fallback `| nothing`. So `python: context.text` should do as a condition. If the textfield is empty that will be `None`.